Feature/druid

zzet opened the pull request.

zzet committed:
Add poseidon gem for communicationswith kafka

zzet committed:
Example (how send events data to druid)

zzet committed:
Fit typo

zzet committed:
Fix hound

zzet committed:
Move business logic from background job to command

vlaew commented on app/commands/process_raw_event.rb:
```diff
@@ -0,0 +1,131 @@
+        memo[:finish] = row[:finish]
+        memo[:duration] = ((memo[:finish] - memo[:start]) * 1000)
+        details.insert(position, memo) if position
+        position = nil
```
А memo тут не надо заодно с position ресетить? Конечно значения в memo перезаписываются, но если вдруг добавится что-то типа такого: memo[:smth] = row[:smth] if some_condition, то будет нехорошо.

zzet replied:
@vlaew где ж ты раньше то был? https://github.com/lenshq/lens_server/pull/31/files#diff-077d447e3b5e1a967043373723857d67R56 :)

zzet committed:
Fix tests and code

zzet committed:
Fix houndci suggestions

zzet committed:
Fix kafka host

jbmeerkat commented on app/commands/process_raw_event.rb:
```diff
@@ -0,0 +1,134 @@
+    scenario_hash = generate_scenario_hash(details)
+
+    event_source = find_or_create_event_source(application: application, meta: meta)
+    scenario = find_or_create_scnario(event_source: event_source, hash: scenario_hash)
```
@zzet опечатка в find_or_create_sc**e**nario

jbmeerkat committed:
extract events persistance code

jbmeerkat committed:
refactor raw event processor

jbmeerkat committed:
fix typo

zzet committed:
Example service locator

zzet committed:
Fix error in transaction parsing #21

zzet committed:
Use druid & kafka

zzet committed:
Fix tests

kgorin commented:
:eyes:

jbmeerkat committed:
refactoring

improve store commands interface
reoder methods
move not public methods to private

jbmeerkat committed:
extract kafka message sender to command

jbmeerkat committed:
remove useless methods

jbmeerkat committed:
remove draft code

jbmeerkat committed:
refactor request model

jbmeerkat committed:
fix typo

jbmeerkat committed:
remove unused code

jbmeerkat committed:
remove useless test

kgorin commented on app/commands/process_raw_event.rb:
```diff
@@ -0,0 +1,138 @@
+      ).to_json
+    end
+    Commands::SendToKafka.call events
+  end
```
надо разбить на методы, как минимум `.call request` и `.call events` можно вынести

kgorin commented on app/models/request.rb:
```diff
@@ -1,16 +1,71 @@
+    private
+
+    def query_by_period(from: nil, to: nil)
+      from ||= Time.now.utc - 1.week
```
константу надо вынести куда-нибудь

zzet committed:
Move default graph period into config

zzet committed:
Move parsed raw event data into value object

zzet committed:
Add meta and details methods to raw_event

zzet committed:
Refactoring of process raw event command with new raw_event interface

zzet committed:
Use delegate in raw event for access to meta and details

zzet committed:
Remove promiscuous code
